SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+This would have the advantage of providing a simple way to test the process for migrating native LSST measurement algorithm code to the Astropy ecosystem in a form that is easily validatable by the community.
+Once complete it should then be possible to consider migrating the more advanced LSST algorithms.
+Making them available in this way should aid adoption by the wider community and provide early feedback as to the quality of the algorithms and detect any bugs or anomalous behavior well before commissioning of LSST begins.
+
+\subsection{NDData}
+
+The Astropy \texttt{NDData} class is an implementation of the common astronomy data structure\cite{2015A&C....12..146J} of a data array, mask and uncertainty with attached metadata.
+LSST has an \texttt{Exposure} class (one for each underlying data type) represented by a \texttt{MaskedImage} and structured metadata.
+The class layout of an \texttt{Exposure} is shown in Fig.~\ref{fig:exposure}.
+It would be beneficial if an LSST \texttt{Exposure} could implement the \texttt{NDData} interface.
+The mask in an \texttt{Exposure} is a bit mask and not a Boolean mask and the uncertainties are always represented as variances (although covariance has been discussed).
+These are supportable as sub-classes and variants of them are already available in the Astropy ecosystem.
+One item that is currently missing from the Astropy arena is the notion of a pixel origin.
+In LSST the pixel origin can be used to specify the coordinates of the data array relative to a parent data array, such as allowing the coordinates of image cutouts to reference where they came from in the parent.
+LSST nomenclature is to use \emph{PARENT} to indicate the abstract pixel coordinate and \emph{LOCAL} to indicate that the coordinates are relative to the 0-based origin of the data array itself\footnote{The Starlink nomenclature\cite{2015A&C....12..146J} is \emph{PIXEL} and \emph{GRID} where \emph{GRID} follows the FITS convention of starting at 1.}.
+This attachment of a data array to a parent pixel coordinate system is a critical item and we plan to work with Astropy on making this functionality available to \texttt{NDData} users.
+
+\begin{figure} [t]
+\begin{center}
+\includegraphics[width=\textwidth]{exposure-dependencies}
+\end{center}
+\caption[layers]
+%>>>> use \label inside caption to get Fig. number with \ref{}
+{\label{fig:exposure}
+The LSST classes that are used to define a \texttt{lsst.afw.image.Exposure}.}
+\end{figure}
+
+\section{Suggested modifications to LSST}
+
+In addition to working with the Astropy community, there is also the question of how integrated the LSST software system should be with Astropy.
+This covers the adoption of distribution standards and use of Astropy classes in our codebase.
+These changes will result in a more supportable and sustainable codebase allowing LSST to focus on core deliverables without having to support general infrastructure.
```

Comment:
Add comma after "sustainable".

"without having to support general infrastructure" -> "while spending less time supporting general infrastructure".  Problem isn't going to go away entirely :-)